clients/drcachesim/tracer/tracer.cpp

```diff
@@ -368,7 +368,7 @@ memtrace(void *drcontext, bool skip_size_cap)
     byte *pipe_start, *pipe_end, *redzone;
     bool do_write = true;
     size_t header_size = 0;
-    uint num_refs = 0;
+    size_t current_num_refs = 0;
```

Comment:
Why the name change?  And the type change?  As a count and not a range, I'm not sure size_t fits logically.

Reply:
There's a global num_refs. At a point, I removed this num_refs, compilation passed, and tests failed. I believe the rename helps maintainability.

The type is to avoid a cast - the span variable ends up being size_t.

Reply:
Oh wow, yes, let's fix the shadowing.

I'm not sold on the type: seems confusing to be size_t.

Reply:
OK, typed back to uint